src/analysis/calculus/fderiv_measurable.lean

```diff
@@ -0,0 +1,455 @@
+with a derivative in `K`. -/
+def D (f : E → F) (K : set (E →L[𝕜] F)) : set E :=
+⋂ (e : ℕ), ⋃ (n : ℕ), ⋂ (p ≥ n) (q ≥ n), B f K ((1/2) ^ p) ((1/2) ^ q) ((1/2) ^ e)
+
+lemma is_open_A (L : E →L[𝕜] F) (r ε : ℝ) : is_open (A f L r ε) :=
+begin
+  rw metric.is_open_iff,
+  rintros x ⟨r', r'_mem, hr'⟩,
+  obtain ⟨s, s_gt, s_lt⟩ : ∃ (s : ℝ), r / 2 < s ∧ s < r' := exists_between r'_mem.1,
+  have : s ∈ Ioc (r/2) r := ⟨s_gt, le_of_lt (s_lt.trans_le r'_mem.2)⟩,
+  refine ⟨r' - s, by linarith, λ x' hx', ⟨s, this, _⟩⟩,
+  have B : ball x' s ⊆ ball x r' := ball_subset (le_of_lt hx'),
+  assume y z hy hz,
+  exact hr' y z (B hy) (B hz)
+end
+
+lemma is_open_B (K : set (E →L[𝕜] F)) (r s ε : ℝ) : is_open (B f K r s ε) :=
+by simp [B, is_open_Union, is_open_inter, is_open_A]
+
+lemma A_mono (L : E →L[𝕜] F) (r : ℝ) {ε δ : ℝ} (h : ε ≤ δ) :
+  A f L r ε ⊆ A f L r δ :=
+begin
+  rintros x ⟨r', r'r, hr'⟩,
+  refine ⟨r', r'r, λ y z hy hz, _⟩,
+  apply le_trans (hr' y z hy hz),
+  apply mul_le_mul_of_nonneg_right h,
+  linarith [mem_ball.1 hy, r'r.2, @dist_nonneg _ _ y x],
+end
+
+lemma le_of_mem_A {r ε : ℝ} {L : E →L[𝕜] F} {x : E} (hx : x ∈ A f L r ε)
+  {y z : E} (hy : y ∈ closed_ball x (r/2)) (hz : z ∈ closed_ball x (r/2)) :
+  ∥f z - f y - L (z-y)∥ ≤ ε * r :=
+begin
+  rcases hx with ⟨r', r'mem, hr'⟩,
+  exact hr' _ _ (lt_of_le_of_lt (mem_closed_ball.1 hy) r'mem.1)
+    (lt_of_le_of_lt (mem_closed_ball.1 hz) r'mem.1)
+end
+
+lemma mem_A_of_differentiable {ε : ℝ} (hε : 0 < ε) {x : E} (hx : differentiable_at 𝕜 f x) :
+  ∃ R > 0, ∀ r ∈ Ioo (0 : ℝ) R, x ∈ A f (fderiv 𝕜 f x) r ε :=
+begin
+  have := hx.has_fderiv_at,
+  simp only [has_fderiv_at, has_fderiv_at_filter, is_o_iff] at this,
+  rcases eventually_nhds_iff_ball.1 (this (half_pos hε)) with ⟨R, R_pos, hR⟩,
+  refine ⟨R, R_pos, λ r hr, _⟩,
+  have : r ∈ Ioc (r/2) r := ⟨half_lt_self hr.1, le_refl _⟩,
+  refine ⟨r, this, λ y z hy hz, _⟩,
+  calc  ∥f z - f y - (fderiv 𝕜 f x) (z - y)∥
+      = ∥(f z - f x - (fderiv 𝕜 f x) (z - x)) - (f y - f x - (fderiv 𝕜 f x) (y - x))∥ :
+    by { congr' 1, simp only [continuous_linear_map.map_sub], abel }
+  ... ≤ ∥(f z - f x - (fderiv 𝕜 f x) (z - x))∥ + ∥f y - f x - (fderiv 𝕜 f x) (y - x)∥ :
+    norm_sub_le _ _
+  ... ≤ ε / 2 * ∥z - x∥ + ε / 2 * ∥y - x∥ :
+    add_le_add (hR _ (lt_trans (mem_ball.1 hz) hr.2)) (hR _ (lt_trans (mem_ball.1 hy) hr.2))
+  ... ≤ ε / 2 * r + ε / 2 * r :
+    add_le_add
+      (mul_le_mul_of_nonneg_left (le_of_lt (mem_ball_iff_norm.1 hz)) (le_of_lt (half_pos hε)))
+      (mul_le_mul_of_nonneg_left (le_of_lt (mem_ball_iff_norm.1 hy)) (le_of_lt (half_pos hε)))
+  ... = ε * r : by ring
+end
+
+lemma norm_sub_le_of_mem_A {c : 𝕜} (hc : 1 < ∥c∥)
+  {r ε : ℝ} (hε : 0 < ε) (hr : 0 < r) {x : E} {L₁ L₂ : E →L[𝕜] F}
+  (h₁ : x ∈ A f L₁ r ε) (h₂ : x ∈ A f L₂ r ε) : ∥L₁ - L₂∥ ≤ 4 * ∥c∥ * ε :=
+begin
+  apply op_norm_le_of_shell (half_pos hr)
+    (mul_nonneg (mul_nonneg (by norm_num : (0 : ℝ) ≤ 4) (norm_nonneg _)) (le_of_lt hε)) hc,
```

Comment:
Could you please write `have : 0 ≤ ...`, `apply op_norm_le_of_shell (half_pos hr) this hc`?

Reply:
Done